### PR TITLE
Figure.grdimage: Remove the unsupported 'img_out'/'A' parameter

### PR DESCRIPTION
**Pull request by seisman**
As proposed in https://github.com/GenericMappingTools/pygmt/pull/2765#issuecomment-1864131592, we will NOT support the `img_out`/`A` parameter in the `Figure.grdimage` method. The related feature will be added to a new function/method if requested.

Supersede #2765.

### CODE REVIEW

File: pygmt/tests/test_grdimage.py [hunk 1]
```diff
@@ -241,3 +241,14 @@ def test_grdimage_central_meridians_and_standard_parallels(grid, proj_type, lon0
     )
     fig_test.grdimage(grid, projection=f"{proj_type}{lon0}/{lat0}/15c", cmap="geo")
     return fig_ref, fig_test
+
+
+def test_grdimage_imgout(grid):
```
Comment (weiji14) [inline pygmt/tests/test_grdimage.py:246]:
```suggestion
def test_grdimage_imgout_fails(grid):
```

File: pygmt/src/grdimage.py [hunk 2]
```diff
@@ -172,6 +157,13 @@ def grdimage(self, grid, **kwargs):
     """
     kwargs = self._preprocess(**kwargs)
 
+    # Do not support -A option
+    if any(kwargs.get(arg) is not None for arg in ["A", "img_out"]):
```
Comment (weiji14) [inline pygmt/src/grdimage.py:161]:
```suggestion
    if any(kwargs.get(arg) for arg in ["A", "img_out"]):
```

Should be ok to remove `is not None`, e.g.:

```python
kwargs = {"img_out": "temp.nc"}
if any(kwargs.get(arg) for arg in ["A", "img_out"]):
    raise Exception
```
should raise an Exception.

**Review by weiji14 — APPROVED** [review]
A couple of minor suggestions, otherwise ok.

File: pygmt/src/grdimage.py (hunk 2 above)
Comment (seisman) [reply pygmt/src/grdimage.py:161]: Won't raise an exception if an empty string is given:
```
kwargs = {"img_out": ""}
if any(kwargs.get(arg) for arg in ["A", "img_out"]):
    raise Exception
```

Comment (weiji14) [reply pygmt/src/grdimage.py:161]: Ah yes, keep it as is then.

**Pull request merged**